### DISCUSSION TITLE
Acesso a firebase no windows.

### DISCUSSION THREAD

**Discussion by pedro0509** [Flutterando/forum#91] [Perguntas e Respostas] [answered]
Olá,
sou novo na linguagem e estou com algumas dúvidas. 

Estou desenvolvendo para desktop e pretendo no futuro levar para mobile, assim é melhor o firebase ou sqlite?

Como eu faço a autenticação de usuário no firebase, sendo que será desktop? Ou então utilizando do sqlite?

Se tiverem material para desktop, ficarei muito agradecido pois na net está bem difícil de achar.

**Comment by WilliansDutra** [comment]
Bom dia.

Cara, essa é a doc oficial:

https://firebase.flutter.dev/docs/overview/

Fiquei com dúvida sobre a sua pergunta então o melhor que posso fazer é te passar esse doc.

**Comment by eduardoflorence** [comment, marked as answer]
Olá @pedro0509,

   A pergunta correta seria Cloud Firestore ou Sqlite, pois o Firebase hoje é uma plataforma com vários serviços, incluindo o banco de dados Firestore. Os dois tem propósitos diferentes. 
   O sqlite armazena dados localmente, então um usuário de seu app não conseguiria compartilhar os dados entre a versão desktop e uma versão mobile.
   O Firestore é um banco em nuvem e seria o ideal, mas não existe ainda um package oficial para ele para desktop, somente o core e o auth, mesmo assim em versão **[beta](https://github.com/invertase/flutterfire_desktop)** há muito tempo e não evolui, então não é seguro agora apostar nele para uma aplicação que você pretenda colocar em produção. 
   O que você pode fazer para usar o Firestore é via backend, criando uma API feita em Cloud Functions, que também é um dos serviços do Firebase. No seu App Flutter é só fazer uma chamada http para cada function que criar. Aqui nesse **[link](https://firebase.google.com/docs/functions/get-started)** dá para você ver um exemplo com a função addMessage.
    Em resumo: Estude o [Cloud Firestore](https://firebase.google.com/docs/firestore) e o [Cloud Functions](https://firebase.google.com/docs/functions)
    Obs.: Até mesmo a autenticação você conseguirá fazer via Cloud Functions

**Comment by pedro0509** [comment]
Eu encontrei alguns tutoriais e uns packages mostrando como colocar o Firestore e o auth no windows, mas estão bem simples e confusos. 
Um dos packages é o firedart 0.9.0+1.

Acredito que por enquanto vou usar sqlite e com a evolução do app eu uso o firebase. 

Muito obrigado pela ajuda de vocês.
